src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/CreateTableRequestFactory.java

```diff
@@ -31,8 +31,6 @@
     /*
      * Returns a list of CreateTableRequests that will be created when the factory is initialized.
      */
-    default List<CreateTableRequest> precreateTables() {
-        return new ArrayList<>();
-    }
+    List<CreateTableRequest> getPhysicalTables();
```

Comment:
It may make sense to store the list of physical tables in MtAmazonDynamoDbBySharedTable and either turn this back into a functional interface (with only the table mapping method) or make table mapping the responsibility of the TableMapper and delete this interface.